### PR TITLE
AAB enabled in the app

### LINKED ISSUE #401: Build app bundles along with apks
**Actual Behaviour.**  
Current APK file size is over 40MB. This will be somewhat problematic when users are downloading.

**Expected Behaviour**  
Using Android app bundle packaging (.aab) file size can be reduced.

**Steps to reproduce it**  
Check apk file sizes in apk branch

**Additional context**  
> Not available

**Would you like to work on the issue?**
Issue for GSoC

### PR DESCRIPTION
**Pull request by jddeep**
Fixes #401 

**Changes**: AAB enabled in the app

**Screenshot/s for the changes**: 
N/A

**Checklist**: [Please tick following check boxes with `[x]` if the respective task is completed]
- [x] I have used resources from `strings.xml`, `dimens.xml` and `colors.xml` without hard-coding them
- [x] No modifications done at the end of resource files `strings.xml`, `dimens.xml` or `colors.xml`
- [x] I have reformatted code in every file included in this PR [<kbd>CTRL</kbd>+<kbd>ALT</kbd>+<kbd>L</kbd>]
- [x] My code does not contain any extra lines or extra spaces
- [x] I have requested reviews from other members

**APK for testing**: 
N/A

### CODE REVIEW

**Review by adityastic — CHANGES_REQUESTED** [review]
Change the scripts and not the bundle configuration. Use badge magic for reference

**Review by adityastic — COMMENTED** [review]
@CloudyPadmal Please make sure that fastlane is initialised in this repository

File: app/build.gradle [hunk 1]
```diff
@@ -37,6 +37,17 @@ android {
         disable 'InvalidPackage'
         disable 'RestrictedApi'
     }
+    bundle {
```
Comment (adityastic) [inline app/build.gradle:40]: This can be removed completely

File: scripts/upload-apk.sh [hunk 2]
```diff
@@ -6,7 +6,9 @@ export DEVELOPMENT_BRANCH=${DEVELOPMENT_BRANCH:-development}
 
 # Setup git
 git config --global user.email "noreply@travis.com"
-git config --global user.name "Travis CI" 
+git config --global user.name "Travis CI"
+
+bundle exec fastlane buildAAB
```
Comment (adityastic) [inline scripts/upload-apk.sh]: This won't work without you installing fastlane as a bundler

File: scripts/upload-apk.sh [hunk 3]
```diff
@@ -61,6 +63,6 @@ git push origin apk -f --quiet > /dev/null
 # TODO: Prepare when a release is ready
 # Publish App to Play Store
 #if [ "$TRAVIS_BRANCH" == "$PUBLISH_BRANCH" ]; then
-    #gem install fastlane
-    #fastlane supply --aab neurolab-master-app.aab --skip_upload_apk true --track alpha --json_key ../scripts/fastlane.json --package_name $PACKAGE_NAME
+    gem install fastlane
+    fastlane supply --aab neurolab-master-app.aab --skip_upload_apk true --track alpha --json_key ../scripts/fastlane.json --package_name $PACKAGE_NAME
```
Comment (adityastic) [inline scripts/upload-apk.sh]: You need to copy it before using it

File: fastlane/Fastfile [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+  desc "Runs all the tests"
+  lane :test do
+    gradle(task: "test")
+  end
+
+  lane :buildAAB do
+    gradle(task: "bundle",build_type: "Release")
+  end
+
+  desc "Submit a new Beta Build to Crashlytics Beta"
```
Comment (adityastic) [inline fastlane/Fastfile]: Remove the rest of stuff

File: fastlane/Appfile [hunk 5]
```diff
@@ -0,0 +1,2 @@
+json_key_file("") # Path to the json secret file - Follow https://docs.fastlane.tools/actions/supply/#setup to get one
```
Comment (adityastic) [inline fastlane/Appfile]: Add the json file path, fastlane needs access to it

**Review by CloudyPadmal — CHANGES_REQUESTED** [review]
We'll add fastlane in a later stage. Just build the appbundle for now.

**Comment by adityastic** [comment]
@CloudyPadmal he updated the pr with the required files. That's all we need as of now

File: scripts/upload-apk.sh [hunk 6]
```diff
@@ -14,6 +14,9 @@ if [ "$TRAVIS_PULL_REQUEST" != "false" -o "$TRAVIS_REPO_SLUG" != "fossasia/neuro
     exit 0
 fi
 
+# Generate Playstore bundle
+./gradlew bundlePlaystoreRelease
```
Comment (CloudyPadmal) [inline scripts/upload-apk.sh:18]: Copy pasted from PSLab right away I guess. This is invalid. Change to `./gradlew bundleRelease`

File: app/build.gradle (hunk 1 above)
Comment (CloudyPadmal) [inline app/build.gradle:40]: Not necessary. Remove the whole block.

**Review by CloudyPadmal — CHANGES_REQUESTED** [review]
Only `./gradlew bundleRelease` is needed in this PR and modify `upload-apk.sh` script to upload that file to apk branch and add it to git lfs.

**Comment by CloudyPadmal** [comment]
Fixed in #428

**Pull request closed without merging**